dask/bag/core.py

```diff
@@ -351,7 +351,17 @@ def __dask_keys__(self):
     def __dask_tokenize__(self):
         return self.key
 
-    __dask_optimize__ = globalmethod(optimize, key="bag_optimize", falsey=dont_optimize)
+    @staticmethod
+    def __dask_optimize__(dsk, keys, **kwargs):
+        if "bag_optimize" in config.config:  # legacy value
+            func = config.get("bag_optimize")
+        else:
+            func = config.get("bag.optimization.function", dont_optimize)
```

Comment:
I think you meant:

```suggestion
            func = config.get("bag.optimization.function", optimize)
```

However, I might be wrong, but I don't think you'll ever need the default for `.get` here. Dask should always load its own config file, which has the defaults already, and they should be there unless they're overwritten with a new value.

Reply:
Thanks, the reason I leave the default value in here is that currently `dask[dataframe]/dask[array]` don't currently require the `yaml` library, so we can't depend on having the config loaded from defaults the same way that we do on every package that also depends on `distributed`.  We could change this though.

Reply:
Ah, I forgot about that. I think leaving the YAML dependency out for now makes sense. Can change later if needed.